### PR TITLE
Suit storage adjustments.

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes the Suit Storage Unit circuitboards researchable, adds stock parts to the Suit Storage Unit. Makes the burn damage of the decontamination process scale with the tier of the micro laser. Also adds the ablity to crowbar open depowered suit storage units.

## Why It's Good For The Game
It's nice to have more roundstart machinery that's replicable, and that also that uses machine parts. Also getting your items out of a depowered suit storage unit is a pain since you need to smash it apart, which takes a long while. This PR remedies that.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Suit Storage Unit research : 

image

image

Prying open a depowered suit storage and prying its contents out :

https://user-images.githubusercontent.com/110184118/215284643-e438fb36-1d22-4bbf-84e4-5f6edabf4da8.mp4

Non-hacked decontamination damage with stock parts : 
image

With t4 parts :
image

Hacked/emagged decontamination damage with stock parts :

image

With t4 parts :

image

</details>

## Changelog
:cl:
add: Added research for a Suit Storage Unit circuit board
add: Added the ability to build and deconstruct Suit Storage Units
add: Added machine parts to the Suit Storage Unit
add: Added the ability to crowbar open and pry the items out of a depowered Suit Storage Unit on harm intent
balance: Suit storage unit's decontamination damage is lower by default when unhacked
/:cl:

### CODE REVIEW

**Comment by Fronsis** [comment]
Nice having them as a craftable is neat tbh!

Do higher-parts mean the rad decontamination % increases? Because as a CMO i've often used it on Engineers who don't want to stand on a shower and chug a beaker for 30 minutes when they have 6000 rads but if they come with 50k it can take a couple of decontamination tries.

Example for my question:
Dude comes with 10k rads

If i ask for my CMO decontainment unit to be upgraded, do the rad-decontamination value goes up?(for example let's say based is 2k and t4 is 4k thus it will take less decontaminaton uses to clear the rads out of them)

**Comment by CydiaLamiales** [comment]
> Nice having them as a craftable is neat tbh!
> 
> Do higher-parts mean the rad decontamination % increases? Because as a CMO i've often used it on Engineers who don't want to stand on a shower and chug a beaker for 30 minutes when they have 6000 rads but if they come with 50k it can take a couple of decontamination tries.
> 
> Example for my question: Dude comes with 10k rads
> 
> If i ask for my CMO decontainment unit to be upgraded, do the rad-decontamination value goes up?(for example let's say based is 2k and t4 is 4k thus it will take less decontaminaton uses to clear the rads out of them)

That's odd, the decontamination should completely remove the radiation
image

**Comment by Fronsis** [comment]
> That's odd, the decontamination should completely remove the radiation. image Also rads get removed at the end of the procedure, not gradually. Hmm, I could maybe speed it up depending on the matter bin level?

Hmm the last time i tried it it was with a player who had probably between 30 or 60k rads and it only removed a couple of thousands of rads for each attempt and another player who was with me told me that it had a cap on how much it rad it took for each use, so we gave up since it was a lot of rads and the irradiated player already died, i guess it got bugged  (unless somehow i hacked it incorrectly?)

Alas if it works now then that's nice! The speed it up could be a good mini encouragment to look for the upgrades.

**Comment by CydiaLamiales** [comment]
> > That's odd, the decontamination should completely remove the radiation. image Also rads get removed at the end of the procedure, not gradually. Hmm, I could maybe speed it up depending on the matter bin level?
> 
> Hmm the last time i tried it it was with a player who had probably between 30 or 60k rads and it only removed a couple of thousands of rads for each attempt and another player who was with me told me that it had a cap on how much it rad it took for each use, so we gave up since it was a lot of rads and the irradiated player already died, i guess it got bugged (unless somehow i hacked it incorrectly?)
> 
> Alas if it works now then that's nice! The speed it up could be a good mini encouragment to look for the upgrades.

Oh, I know why. If the safeties get disabled and the strength gets increased while hacking then rads won't get removed. Don't know if it's intentional or not, but this PR makes it null the rads no matter what.

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -42,6 +41,10 @@
 	* * If FALSE, decontamination sequence will clear radiation for all atoms (and their contents) contained inside the unit, and burn any mobs inside.
 	* * If TRUE, decontamination sequence will burn and decontaminate all items contained within, and if occupied by a mob, intensifies burn damage delt. All wires will be cut at the end.
 	*/
+	///how strong is the burn damage during cook(), decreases with micro laser tier
+	var/laser_strength = 10
+	///how strong is the burn damage if hacked/emagged during cook(), increases with micro laser tier
+	var/laser_strength_hacked =  20
```
Comment (itsmeow) [inline code/game/machinery/suit_storage_unit.dm]: there's a space here that must go

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -199,6 +210,12 @@
 	storage = null
 	occupant = null
 
+/obj/machinery/suit_storage_unit/proc/is_empty()
+	var empty = FALSE
+	if(isnull(helmet) && isnull(suit) && isnull(mask) && isnull(storage) && isnull(occupant))
+		empty = TRUE
+	return empty
```
Comment (itsmeow) [inline code/game/machinery/suit_storage_unit.dm]: no need for the var, just return the expression

File: code/game/machinery/suit_storage_unit.dm [hunk 3]
```diff
@@ -258,16 +278,16 @@
 */
 /obj/machinery/suit_storage_unit/proc/cook()
 	var/mob/living/mob_occupant = occupant
+	var/burn_damage = laser_strength
+	if(uv_super || (obj_flags & EMAGGED))
```
Comment (itsmeow) [inline code/game/machinery/suit_storage_unit.dm]: you could use a ternary here `var/burn_damage = uv_super || (obj_flags & EMAGGED) ? laser_strength_hacked : laser_strength`

File: code/game/machinery/suit_storage_unit.dm [hunk 4]
```diff
@@ -278,17 +298,18 @@
...
+				helmet.take_damage(burn_damage * 10,BURN,"fire")
 			if(suit)
-				suit.take_damage(100,BURN,"fire")
+				suit.take_damage(burn_damage * 10,BURN,"fire")
 			if(mask)
-				mask.take_damage(100,BURN,"fire")
+				mask.take_damage(burn_damage * 10,BURN,"fire")
 			if(storage)
-				storage.take_damage(100,BURN,"fire")
+				storage.take_damage(burn_damage * 10,BURN,"fire")
```
Comment (itsmeow) [inline code/game/machinery/suit_storage_unit.dm:307]: this should also probably remove the rads

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by itsmeow — APPROVED** [review]

**Pull request merged**